File: fastapi-python-backend/tests/metrics_test.py
from fastapi.testclient import TestClient


def test_returns_prometheus_metrics_payload(client: TestClient) -> None:
    response = client.get('/metrics')

    assert response.status_code == 200
    assert 'text/plain' in response.headers['content-type']
    assert 'http_requests_total' in response.text
    assert 'http_request_duration_seconds' in response.text

File: fastify-ts-bun-backend/tests/metrics.test.ts
import type { FastifyInstance } from 'fastify';
import { afterEach, describe, expect, it } from 'vitest';
import { createApp } from '../src/app';

describe('GET /metrics', () => {
  let app: FastifyInstance | undefined;

  afterEach(async () => {
    if (app) {
      await app.close();
      app = undefined;
    }
  });

  it('returns Prometheus metrics payload', async () => {
    app = await createApp();
    const instance = app;

    const response = await instance.inject({
      method: 'GET',
      url: '/metrics',
    });

    expect(response.statusCode).toBe(200);
    expect(response.headers['content-type']).toContain('text/plain');
    expect(response.body).toContain('http_requests_total');
    expect(response.body).toContain('http_request_duration_seconds');
  });
});
